### PR TITLE
Back-end bonanza: Create server, mongoose models and connect database

### PR DESCRIPTION
**Pull request by batmanimal**
@csaden to review and merge

[Review on Reviewable](https://reviewable.io/reviews/agate-rain/gifty/10)

### CODE REVIEW

File: server/giftlists/giftlistModel.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+var Schema = require('mongoose').Schema;
+
+var GiftListSchema = new Schema({
+  userId: {
+    type: ObjectId,
+    ref: 'User'
+  },
+  // giftee
+  friendId: {
+    type: ObjectId,
```
Comment (csaden) [inline server/giftlists/giftlistModel.js]: I think type should be `Schema.Types.ObjectId` when we are using a ref

File: server/giftlists/giftlistModel.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    ref: 'User'
+  },
+  gifts: [{
+    type: ObjectId,
```
Comment (csaden) [inline server/giftlists/giftlistModel.js]: `Schema.Types.ObjectId` here too

File: server/giftlists/giftlistModel.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+var model = require('mongoose').model;
+var Schema = require('mongoose').Schema;
+
+var GiftListSchema = new Schema({
+  userId: {
+    type: ObjectId,
```
Comment (csaden) [inline server/giftlists/giftlistModel.js]: `Schema.Types.ObjectId` here

File: server/users/userModel.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+var model = require('mongoose').model;
+var Schema = require('mongoose').Schema;
+
+// represents all users of the app, both gifters and giftees 
+var UserSchema = new Schema({
+  birthdate: {
+    // may need to change to Date, using String bc we don't know what FB will give us
+    type: String,
+    required: true
```
Comment (csaden) [inline server/users/userModel.js]: We should remove this as being required. Some users might not have birthdays from FB. We should prompt the users to save their birthday during signup flow and require a valid birthday.

**Comment by batmanimal** [comment]
updated and will create new PR now! thank you!

**Comment by batmanimal** [comment]
@csaden Pushed those changes into this PR -- now ready for your eyeballs!

**Pull request merged**